app/src/main/java/com/sunasterisk/a14day_challenge/ui/signup/SignUpPresenter.kt

```diff
@@ -0,0 +1,48 @@
+package com.sunasterisk.a14day_challenge.ui.signup
+
+import com.sunasterisk.a14day_challenge.R
+import com.sunasterisk.a14day_challenge.data.UserRepository
+import com.sunasterisk.a14day_challenge.data.model.User
+
+class SignUpPresenter(
+    var view: SignUpContract.View,
+    private val userRepository: UserRepository
+) : SignUpContract.Presenter {
+
+    override fun handleSignUp(
+        confirmPassword: String,
+        username: String,
+        name: String,
+        password: String,
+        birthday: String,
+        height: String,
+        weight: String
+    ) {
+        val user = User(
+            username,
+            name,
+            password,
+            birthday,
+            height,
+            weight,
+            process = 1
+        )
+        var isExistedUser = false
+        val userList = userRepository.getAllUsers()
+        userList?.forEach {
+            if (user.account.contentEquals(it.account)) {
+                isExistedUser = true
+            }
+        }
+        if (user.account.isEmpty() || user.password.isEmpty() || user.name.isEmpty()) {
+            view.onSignUpFailure(R.string.error_signup_empty.toString())
+        } else if (!user.password.contentEquals(confirmPassword)) {
+            view.onSignUpFailure(R.string.error_confirm_password.toString())
+        } else if (isExistedUser) {
+            view.onSignUpFailure(R.string.error_existed_account.toString())
+        } else {
+            userRepository.addUser(user)
+            view.onSignUpSucceeded()
+        }
```

Comment:
không nên để if else lồng nhau như này 
Mà đoạn `R.string.error_existed_account.toString()` sẽ hiển thị cái gì thế 
```suggestion
        if (user.account.isEmpty() || user.password.isEmpty() || user.name.isEmpty()) {
            view.onSignUpFailure(R.string.error_signup_empty.toString())
            return 
        } 
        if (!user.password.contentEquals(confirmPassword)) {
            view.onSignUpFailure(R.string.error_confirm_password.toString())
            return 
        } 
        if (isExistedUser) {
            view.onSignUpFailure(R.string.error_existed_account.toString())
            return 
        } 
        
            userRepository.addUser(user)
            view.onSignUpSucceeded()
        }
```